linux/vhba_hot_add_remove/freebsd_create_disk_partition.yml

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022-2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   Format and create 1 new partition on a disk
+# Parameters:
+#   disk_name: The disk name to be formatted and create partition
+#   partition_fstype: Create filesystem on new partition, which could be ext4, xfs
+#                         or btrfs, etc. The default filesystem is ext4.
+# Return:
+#   partition_name: The disk partition name
+#   partition_device_path: The disk partition device path
+#
+- name: "Initialize disk partition name and device path"
+  ansible.builtin.set_fact:
+    partition_name: ""
+    partition_device_path: ""
+
+- name: "Set default partition filesystem to ext4"
+  ansible.builtin.set_fact:
+    partition_fstype: "ufs"
+  when: >
+    partition_fstype is undefined or
+    not partition_fstype
+
+- name: "Filesystem {{ partition_fstype }} is not supported"
+  ansible.builtin.fail:
+    msg: >-
+      Guest OS doesn't support Filesystem {{ partition_fstype }}, so it can't
+      create partition with filesystem {{ partition_fstype }} on disk {{ disk_name }}
+  when: partition_fstype is not in ['ufs', 'zfs']
+
+- name: "Create partition table on disk {{ disk_name }}"
+  ansible.builtin.shell: gpart create -s gpt {{ disk_name }}
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Create 1 partition on disk {{ disk_name }}"
+  ansible.builtin.shell: gpart add -t freebsd-{{ partition_fstype }} {{ disk_name }}
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Set fact of partition name"
+  ansible.builtin.set_fact:
+    partition_name: "{{ disk_name }}p1"
+
+- name: "Set fact of partition device path"
+  ansible.builtin.set_fact:
+    partition_device_path: "/dev/{{ partition_name }}"
+
+- name: "Create {{ partition_fstype }} filesystem on partition {{ partition_device_path }}"
+  ansible.builtin.command: "newfs -U {{ partition_device_path }}"
+  delegate_to: "{{ vm_guest_ip }}"
+  ignore_errors: true
+  register: partition_result
+  when: partition_fstype == "zfs"
+
+- name: "Create {{ partition_fstype }} filesystem on partition {{ partition_device_path }}"
+  community.general.filesystem:
+    fstype: "{{ partition_fstype }}"
+    dev: "{{ partition_device_path }}"
+  delegate_to: "{{ vm_guest_ip }}"
+  register: partition_result
+  when: partition_fstype == "ufs"
+
+- name: Display the partiton creation result
+  ansible.builtin.debug: var=partition_result
```

Comment:
The debug partition_result will always print the result of   the one registered at line 61. So it is meaningless to print it here.

Reply:
Remove this task. Thanks